src/quo/components/links/internal_link_card/view.cljs

```diff
@@ -0,0 +1,14 @@
+(ns quo.components.links.internal-link-card.view
+  (:require [quo.components.links.internal-link-card.channel.view :as channel.view]
+            [quo.components.links.internal-link-card.community.view :as community.view]
+            [quo.components.links.internal-link-card.user.view :as user.view]
+            [quo.theme :as quo.theme]))
+
+(defn- internal-link-card
+  [{card-type :type :as args}]
+  (case card-type
+    :community [community.view/view args]
+    :channel   [channel.view/view args]
+    :user      [user.view/view args]))
+
+(def view (quo.theme/with-theme internal-link-card))
```

Comment:
you don't need to wrap with `with-theme` here. This is only needed when you need access to `theme` in the direct component, which you don't use here. 👍